src/core/client_channel/direct_channel.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2024 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/client_channel/direct_channel.h"
+
+#include "src/core/lib/config/core_configuration.h"
+#include "src/core/lib/gprpp/orphanable.h"
+#include "src/core/lib/surface/channel_stack_type.h"
+#include "src/core/lib/surface/client_call.h"
+#include "src/core/lib/transport/interception_chain.h"
+
+namespace grpc_core {
+
+absl::StatusOr<RefCountedPtr<DirectChannel>> DirectChannel::Create(
+    std::string target, const ChannelArgs& args) {
+  auto* transport = args.GetObject<Transport>();
+  if (transport == nullptr) {
+    return absl::InvalidArgumentError("Transport not set in ChannelArgs");
+  }
+  if (transport->client_transport() == nullptr) {
+    return absl::InvalidArgumentError("Transport is not a client transport");
+  }
+  auto transport_call_destination = MakeRefCounted<TransportCallDestination>(
+      OrphanablePtr<ClientTransport>(transport->client_transport()));
+  auto event_engine =
+      args.GetObjectRef<grpc_event_engine::experimental::EventEngine>();
+  if (event_engine == nullptr) {
+    return absl::InvalidArgumentError("EventEngine not set in ChannelArgs");
+  }
+  InterceptionChainBuilder builder(args);
+  CoreConfiguration::Get().channel_init().AddToInterceptionChainBuilder(
+      GRPC_CLIENT_DIRECT_CHANNEL, builder);
+  auto interception_chain =
+      builder.Build(std::move(transport_call_destination));
+  if (!interception_chain.ok()) {
+    return interception_chain.status();
+  }
+  return MakeRefCounted<DirectChannel>(
+      std::move(target), args, std::move(event_engine),
+      std::move(transport_call_destination), std::move(*interception_chain));
+}
+
+void DirectChannel::Orphaned() {
+  transport_call_destination_.reset();
+  interception_chain_.reset();
+}
+
+void DirectChannel::StartCall(UnstartedCallHandler unstarted_handler) {
+  unstarted_handler.SpawnInfallible(
+      "start", [this, unstarted_handler]() mutable {
+        interception_chain_->StartCall(std::move(unstarted_handler));
+        return []() { return Empty{}; };
+      });
+}
+
+void DirectChannel::GetInfo(const grpc_channel_info* info) {}
```

Comment:
It looks like this no-op is actually the same behavior as the legacy stack, so it's fine for now, but I think we should be able to do better.  Please add a TODO(roth) about having this return the default service config, if any.